File: temp/html_to_pdf.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
  const inputPath = 'D:\\新课开发\\经济学\\11_全球化与国际收支\\可打印工具卡\\03-汇率决定理论速查卡.html';
  const outputPath = 'D:\\新课开发\\经济学\\11_全球化与国际收支\\可打印工具卡\\03-汇率决定理论速查卡.pdf';

  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Set viewport to A4
  await page.setViewportSize({ width: 794, height: 1123 }); // 210mm x 297mm at 96dpi

  // Load the HTML
  await page.goto('file:///' + inputPath.replace(/\\/g, '/'), { waitUntil: 'networkidle' });

  // Generate PDF with print settings
  await page.pdf({
    path: outputPath,
    width: '210mm',
    height: '297mm',
    printBackground: true,
    margin: { top: 0, right: 0, bottom: 0, left: 0 }
  });

  await browser.close();
  console.log('PDF generated:', outputPath);
})();

File: temp/html_to_pdf.py
from playwright.sync_api import sync_playwright
import os

html_path = "D:/新课开发/HR/员工关系/8.管理者角色升级-从执行者到人性化管理者的能力重塑/12-可打印工具卡/工具卡合集.html"
pdf_path = "D:/新课开发/HR/员工关系/8.管理者角色升级-从执行者到人性化管理者的能力重塑/12-可打印工具卡/工具卡合集.pdf"

# Convert relative path to absolute file URL
html_path_abs = os.path.abspath(html_path)
html_url = f"file:///{html_path_abs.replace(chr(92), '/')}"

print(f"HTML path: {html_path}")
print(f"HTML URL: {html_url}")
print(f"PDF path: {pdf_path}")

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()

    print("Loading HTML...")
    page.goto(html_url)

    # Wait for content to load
    page.wait_for_load_state("networkidle")

    print("Generating PDF...")
    page.pdf(
        path=pdf_path,
        format="A4",
        print_background=True,
        margin={"top": "8mm", "right": "8mm", "bottom": "8mm", "left": "8mm"}
    )

    browser.close()

print(f"PDF created: {pdf_path}")

# Verify file size
if os.path.exists(pdf_path):
    size = os.path.getsize(pdf_path)
    print(f"PDF size: {size / 1024:.1f} KB")
else:
    print("PDF was not created")
